insights/parsers/ldif_config.py

```diff
@@ -0,0 +1,106 @@
+"""
+LDIF Configuration - file ``/etc/dirsrv/slapd-*/dse.ldif``
+==========================================================
+"""
+
+from insights import Parser, parser
+from insights.specs import Specs
+from insights.parsers import SkipException
+import re
+
+
+@parser(Specs.ldif_config)
+class LDIFParser(Parser, list):
+    """
+    Parse the content of the directory server configuration from
+    the ``/etc/dirsrv/slapd-*/dse.ldif`` file.
+
+    The file dse.ldif is in the LDIF format. LDIF contains multi-row records
+    where each record is identified by a ``dn:`` line ("dn" as in "distinguished
+    name") and the record's other lines are attributes. The value may be specified
+    as UTF-8 text or as base64 encoded data, or a URI may be provided to the
+    location of the attribute value.
+
+    Sample output of the ``/etc/dirsrv/slapd-*/dse.ldif`` file::
+
+        dn: cn=changelog5,cn=config
+        cn: changelog5
+        createTimestamp: 20201026161228Z
+        creatorsName: cn=Directory Manager
+        modifiersName: cn=Directory Manager
+        modifyTimestamp: 20201026161228Z
+        nsslapd-changelogdir: /var/lib/dirsrv/slapd-IDM-NYPD-FINEST/cldb
+        nsslapd-changelogmaxage: 7d
+        objectClass: top
+        objectClass: extensibleobject
+
+    Examples:
+        >>> ldif_config[3]['cn=changelog5,cn=config']['modifiersName']
+        'cn=Directory Manager'
+        >>> ldif_config[3]['cn=changelog5,cn=config']['modifyTimestamp']
+        '20201026161228Z'
+    """
+    def parse_content(self, content):
+        if not content:
+            raise SkipException('The file is empty')
+
+        m_flag = 0
+        for line in content:
+            attr_kval = {}
+            # line is emtpy
+            if not line:
```

Comment:
better ensure the line is not a line which only contains `space` before this check.
`line = line.strip()`